File: solutions/c/prime-factors/1/prime_factors.c
#include "prime_factors.h"

size_t find_factors(uint64_t n, uint64_t factors[static MAXFACTORS]) {
    size_t count = 0;
    uint64_t divisor = 2;

    while (divisor * divisor <= n) {
        while (n % divisor == 0) {
            factors[count++] = divisor;
            n /= divisor;
        }
        divisor++;
    }

    if (n > 1) {
        factors[count++] = n;
    }

    return count;
}

File: solutions/javascript/prime-factors/1/prime-factors.js
export const primeFactors = (number) => {
  const factors = [];
  let remaining = number;
  let divisor = 2;

  while (divisor * divisor <= remaining) {
    while (remaining % divisor === 0) {
      factors.push(divisor);
      remaining /= divisor;
    }
    divisor++;
  }

  if (remaining > 1) {
    factors.push(remaining);
  }

  return factors;
};